db/thrman.c

```diff
@@ -587,6 +588,7 @@ void resume_threads(struct dbenv *dbenv)
     if (!dbenv->purge_old_blkseq_is_running ||
         !dbenv->purge_old_files_is_running)
         create_old_blkseq_thread(dbenv);
+    create_watcher_thread(dbenv->bdb_env);
```

Comment:
I think the watcher thread needs to be registered in thrman (using `thrman_register()`) and checked in [`thrman_check_threads_stopped_ll()`](https://github.com/bloomberg/comdb2/blob/master/db/thrman.c#L462-L470). There'd otherwise be a slim chance that we reach here before the watcher thread's[ stop check](https://github.com/bloomberg/comdb2/blob/master/bdb/rep.c#L5275) (since `stop_threads()` doesn't wait for the thread to exit), and wind up having 2 watcher threads.

Reply:
YES .. thank you ..

Reply:
> Mike has a different (better) solution in https://github.com/bloomberg/comdb2/pull/2110 - I'm removing that part.